core/src/epicli/cli/engine/ansible/AnsibleRunner.py

```diff
@@ -152,6 +161,12 @@ def upgrade(self):
         # pre-flight to prepare machines
         self.pre_flight(inventory_path)
 
+        self.ansible_command.run_playbook(inventory=inventory_path,
+                                          playbook_path=self.playbook_path(to_role_name('versioning')),
+                                          extra_vars={'mode': 'upgrade',
+                                                      'date': datetime.datetime.now().ctime(),
+                                                      'version': VERSION})
```

Comment:
There is `is_upgrade_run` fact and date/time can be used from ansible. Is it necessary to pass these parameters and modify `run_playbook ` for that?

Reply:
Yes, there can be also time differences, remote time (on VM) `ansible_date_time` can be different than local time (on Ansible controller).

Reply:
I do not agree with the `is_upgrade_run` change. If we want to add something later we will probably end up with a bunch of `if` `else`s. Here the script doesn't care about which mode it is, so there is no place for conditionals.

Reply:
About `ansible_date_time`, maybe we should add both local and remote time?

Reply:
We don't need to implement `is_upgrade_run`, it's already in place and used in Ansible configuration.

Reply:
I meant that it's a bad design, and if there is a better way of doing it, we should do it. Scripts should not be aware of the mode they are in, this leads to putting conditionals everywhere. In our python code we have distinct paths for both apply and upgrade modes.

Reply:
Ok, before talking about design, please change the code not to use a separate code. Then we will see what’s necessary in addition.